src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -594,7 +625,7 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                     // response, since it's not a valid parameter when using the implicit flow.
                     // See http://openid.net/specs/openid-connect-core-1_0.html#Authentication
                     // and https://tools.ietf.org/html/rfc6749#section-4.2.2.
-                    SaveTokens(ticket.Principal, message, saveRefreshToken: false);
+                    SaveTokens(ticket.Principal, message, jwt.Issuer, saveRefreshToken: false);
```

Comment:
Hey Brent, the idea here is that given that we do verify that those claims are coming from the issuer, we should reflect that in the resulting Claim entries. The mechanics through which we obtained those claims should not make a difference form code that simply consumes the claim values and need to know whether those claims are backed by the issuer.